src/vmd.h

```diff
@@ -1,103 +1,70 @@
-﻿#ifndef VMDTOFBX_VMD_H
-#define VMDTOFBX_VMD_H
-typedef unsigned long DWORD;
-typedef unsigned char BYTE;
-
-#include <cstdio>
-#include <fstream>
-#include <vector>
-#include <map>
-#include <string>
-#include <iostream>
-#include <list>
-using namespace std;
-
-
-struct VMDHeader{
-    char Version[30];
-    char ModelName[20];
-    //unsigned long FrameData[4];
-};
-
-struct VMDMotionFrame{
-    char BoneName[15];
-    DWORD FrameNo;
-    float Location[3];
-    float Rotation[4];
-    BYTE Interpolation[64];
-};
-
-//
-//
-// 表情データ数
-struct VMDMorphHeader {
-    unsigned long Count; // 表情データ数
-} ;
-
-struct VMDMorphFrame {
-    char SkinName[15]; // 表情名
-    unsigned long FrameNo; // フレーム番号
-    float Weight; // 表情の設定値(表情スライダーの値)
-};
-//
-// カメラデータ数
-struct VMD_CAMERA_COUNT {
-    unsigned long Count; // カメラデータ数
-} ;
-
-//// カメラデータ
-//struct VMD_CAMERA { // 61 Bytes // カメラ
-//    unsigned long FrameNo; // フレーム番号
-//    float Length; // -(距離)
-//    float Location[3]; // 位置
-//    float Rotation[3]; // オイラー角 // X軸は符号が反転しているので注意 // 回転
-//    char Interpolation[24]; // おそらく[6][4](未検証) // 補完
-//    unsigned long ViewingAngle; // 視界角
-//    char Perspective; // 0:on 1:off // パースペクティブ
-//} ;
-//
-//// 照明データ数
-//struct VMD_LIGHT_COUNT {
-//    unsigned long Count; // 照明データ数
-//} ;
-//
-//// 照明データ
-//struct VMD_LIGHT { // 28 Bytes // 照明
-//    unsigned long FrameNo; // フレーム番号
-//    float RGB[3]; // RGB各値/256 // 赤、緑、青
-//    float Location[3]; // X, Y, Z
-//} ;
-//
-//// セルフシャドウデータ数
-//struct VMD_SELF_SHADOW_COUNT {
-//    unsigned long Count; // セルフシャドウデータ数
-//} vmd_self_shadow_count;
-//
-//// セルフシャドウデータ
-//struct VMD_SELF_SHADOW { // 9 Bytes // セルフシャドー
-//    unsigned long FrameNo; // フレーム番号
-//    char Mode; // 00-02 // モード
-//    float Distance; // 0.1 - (dist * 0.00001) // 距離
-//};
-//
-//
-
-class VMD{
-public:
-    int BoneCount;
-    int MorphCount;
-    int CameraCount;
-//    int LightCount;
-//    int ShadowCount;
-    VMDHeader Header;
-    std::vector<VMDMotionFrame> *MotionFrames;
-    std::vector<VMDMorphFrame> *MorphFrames;
-
-public :
-    void Read(const char* filePath);
-    std::vector<const char*> GetMorphList();
-
-
-};
-
-#endif //VMDTOFBX_VMD_H
+// VMD format & functions
+// source : https://blog.goo.ne.jp/torisu_tetosuki/e/bc9f1c4d597341b394bd02b64597499d
+
+#ifndef __VMD_H__
+#define __VMD_H__
+
+#include <cstdint>
+#include <fstream>
+#include <vector>
+#include <algorithm>
+
+#pragma pack(push, 1)
+
+static const size_t VMD_HEADER_SIZE = 50;
+static const size_t VMD_MOTION_SIZE = 111;
+
+// 表情データ
+struct VMD_SKIN
+{
+    char SkinName[15]; // Shift_JIS
+    uint32_t FrameNo;
+    float Weight;
+};
```

Comment:
```suggestion
struct VMDSkin
{
    char skin_name[15]; // Shift_JIS
    uint32_t frame_no;
    float weight;
};
```
C++の命名規則([Google C++ スタイルガイド](https://ttsuki.github.io/styleguide/cppguide.ja.html#General_Naming_Rules))に倣うならこの形が望ましいです。
ここを変更する場合、関連するすべての箇所に変更を適用しなければならないことに注意してください...!

Reply:
ここは[VMD仕様の記載元](https://blog.goo.ne.jp/torisu_tetosuki/e/bc9f1c4d597341b394bd02b64597499d)と同じ記述にしていました。
スタイルガイドに沿った書き方が適切だと思いますが、記載元にも倣おうと思っていたので正直迷っています。